internal/services/dataprotection/data_protection_backup_instance_kubernetes_cluster_resource.go

```diff
@@ -0,0 +1,354 @@
+				VaultId: vaultId.ID(),
+			}
+
+			if model := resp.Model; model != nil {
+				if properties := model.Properties; properties != nil {
+					state.Location = location.NormalizeNilable(properties.DataSourceInfo.ResourceLocation)
+					state.BackupPolicyId = properties.PolicyInfo.PolicyId
+					state.KubernetesClusterId = properties.DataSourceInfo.ResourceID
+
+					if policyParameters := properties.PolicyInfo.PolicyParameters; policyParameters != nil {
+						if dataStorePara := policyParameters.DataStoreParametersList; dataStorePara != nil {
+							if dsp := pointer.From(dataStorePara); len(dsp) > 0 {
+								if parameter, ok := dsp[0].(backupinstances.AzureOperationalStoreParameters); ok && parameter.ResourceGroupId != nil {
+									resourceGroupId, err := resourceParse.ResourceGroupID(*parameter.ResourceGroupId)
+									if err != nil {
+										return err
+									}
+									state.SnapshotResourceGroupName = resourceGroupId.ResourceGroup
+								}
+							}
+						}
+						if backupDsp := policyParameters.BackupDatasourceParametersList; backupDsp != nil {
+							if v := flattenBackupDatasourceParameters(*backupDsp); v != nil {
+								state.BackupDatasourceParameters = pointer.From(v)
+							}
+						}
+					}
+				}
+			}
+
+			return metadata.Encode(&state)
+		},
+	}
+}
+
+func (r DataProtectionBackupInstanceKubernatesClusterResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.DataProtection.BackupInstanceClient
+
+			id, err := backupinstances.ParseBackupInstanceID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			err = client.DeleteThenPoll(ctx, *id)
+			if err != nil {
+				return fmt.Errorf("deleting DataProtection Backup Instance (%q): %+v", *id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func expandBackupDatasourceParameters(input []BackupDatasourceParameters) *[]backupinstances.BackupDatasourceParameters {
+	if len(input) == 0 {
+		return nil
+	}
+	results := make([]backupinstances.BackupDatasourceParameters, 0)
+	results = append(results, backupinstances.KubernetesClusterBackupDatasourceParameters{
+		ExcludedNamespaces:           pointer.To(input[0].ExcludedNamespaces),
+		ExcludedResourceTypes:        pointer.To(input[0].ExcludedResourceTypes),
+		IncludeClusterScopeResources: input[0].ClusterScopeResourceEnabled,
+		IncludedNamespaces:           pointer.To(input[0].IncludedNamespaces),
+		IncludedResourceTypes:        pointer.To(input[0].IncludedResourceTypes),
+		LabelSelectors:               pointer.To(input[0].LabelSelectors),
+		SnapshotVolumes:              input[0].VolumeSnapshotEnabled,
+	})
+	return &results
+}
+
+func flattenBackupDatasourceParameters(input []backupinstances.BackupDatasourceParameters) *[]BackupDatasourceParameters {
+	if len(input) == 0 {
+		return nil
+	}
+
+	results := make([]BackupDatasourceParameters, 0)
```

Comment:
We need to set an empty value into state instead of nil.
```suggestion
	results := make([]BackupDatasourceParameters, 0)
	if len(input) == 0 {
		return &results
	}
```

Reply:
Fixed.